packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1580,38 +1580,84 @@ abstract class RenderObject extends AbstractNode with DiagnosticableTreeMixin im
     late bool result;
     assert(() {
       if (_debugDisposed) {
+        assert(false, 'RenderObject $this has already been disposed.');
         result = false;
         return true;
       }
```

Comment:
Well I am a bit confused. If `assert(false)`, how can the lines *after* it be executed? (the `result = false` etc). Especially considering all these are wrapped in `assert((){ })` such that the whole block will only exist in debug build.

Reply:
I'll change that to throw a `FlutterError` then.